source/common/router/config_impl.cc

```diff
@@ -854,5 +876,27 @@ ConfigImpl::ConfigImpl(const envoy::api::v2::RouteConfiguration& config, Runtime
                                                      config.response_headers_to_remove());
 }
 
+PerFilterConfigs::PerFilterConfigs(const Protobuf::Map<std::string, ProtobufWkt::Struct>& configs) {
+  for (const auto& cfg : configs) {
+    const std::string& name = cfg.first;
+    const ProtobufWkt::Struct& struct_config = cfg.second;
+
+    auto& factory = Envoy::Config::Utility::getAndCheckFactory<
+        Server::Configuration::NamedHttpFilterConfigFactory>(name);
+
+    configs_[name] = Envoy::Config::Utility::translateToFactoryRouteConfig(struct_config, factory);
```

Comment:
create another map 

config_hash_ to store hash of config

Reply:
Same comment as above. I don't think this is very helpful and I would rather sort out filter update callbacks as a separate feature?